src/zc/recipe/testrunner/tests.py

```diff
@@ -12,57 +12,166 @@
 #
 ##############################################################################
 
+from zc.buildout.testing import mkdir, write, system
+
 import doctest
+import os
 import re
+import shutil
+import tempfile
 import unittest
-
 import zc.buildout.testing
-import doctest
 import zope.testing.renormalizing
 
 
+SETUP_PY = """from setuptools import setup
+setup(name = "bugfix1")
+"""
+
+
+BUILDOUT_CFG = """[buildout]
+develop = bugfix1
+parts = testbugfix1
+offline = true
+[testbugfix1]
+recipe = zc.recipe.testrunner
+eggs =
+    bugfix1
+script = test
+working-directory = sample_working_dir
+"""
+
+
+TESTS_PY = """import unittest
+class Layer1(object):
+    pass
+class Layer2(object):
+    pass
+class TestDemo1(unittest.TestCase):
+    def test(self):
+        pass
+class TestDemo2(unittest.TestCase):
+    def test(self):
+        pass
+
+def test_suite():
+    suite = unittest.TestSuite()
+    suite1 = unittest.makeSuite(TestDemo1)
+    suite1.layer = Layer1
+    suite.addTest(suite1)
+    suite2 = unittest.makeSuite(TestDemo2)
+    suite2.layer = Layer2
+    suite.addTest(suite2)
+    return suite
+"""
+
+
+OUTPUT_COMP = """Running tests at level 1
+Running .EmptyLayer tests:
+Set up .EmptyLayer
+Running bugfix1.tests.Layer1 tests:
+Running in a subprocess.
+Set up bugfix1.tests.Layer1 in
+Ran 1 tests with 0 failures, 0 errors and 0 skipped in
+Tear down bugfix1.tests.Layer1 in
+Running bugfix1.tests.Layer2 tests:
+Running in a subprocess.
+Set up bugfix1.tests.Layer2 in
+Ran 1 tests with 0 failures, 0 errors and 0 skipped in
+Tear down bugfix1.tests.Layer2 in
+Tearing down left over layers:
+Tear down .EmptyLayer in
+Total: 2 tests, 0 failures, 0 errors and 0 skipped in
+"""
+
+
+class AbsPathTest(unittest.TestCase):
+    """Since version 3.6.0 zope.testing can run layers in parallel
+    processes. zc.recipe.testrunner sets the working directory for each
+    using a relative path. The subprocesses have the given working
+    directory already set, but zc.recipe.testrunner tried to set it again
+    and failed ("IOError: No such file or directory")."""
```

Comment:
PEP-257 recommends moving the closing `"""` to a line of its own, for multi-line docstrings.